qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -779,7 +779,7 @@ def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **
         use_inverted = options.get("use_inverted", False)
         use_transposed = options.get("use_transposed", False)
 
-        mat = high_level_object.linear.astype(int)
+        mat = high_level_object.linear.astype(bool)
```

Comment:
This is already dtype bool in `LinearFunction` so we should avoid a copy here:

```suggestion
        mat = high_level_object.linear.astype(bool, copy=False)
```